### ISSUE TITLE
Differences in RoundHalfUp/Down strategies in different implementations

### ISSUE THREAD

**Issue by symbiont-alexey-babkin** [fpco/safe-decimal#6]
It looks like `RoundHalfUp` in this lib will result in the following rounding `-0.15` -> `-0.1`, where as in some other languages/libs (notable Python, Java, ...) it is `-0.15` -> `-0.2`.
The reason I've stumbled on this is because I'm reimplementing a behavior written in Python and so there was this implicit expectation.
I've dug into this and it seems there is a split consensus on rounding negative decimals: https://en.wikipedia.org/wiki/Rounding#Round_half_up
There seems to be IEEE standard for the rounding: https://en.wikipedia.org/wiki/IEEE_754#Roundings_to_nearest
I do need the same rounding behavior as in Python (to and away from zero) and I imagine some users may need this too. 
I was curious if it would be a lot of work to add these two missing strategies to the library? or whether authors would oppose that. 
Case in point, while DAML does not use `safe-decimal`, they seem to prefer the to/away from zero rounding strategies: https://github.com/digital-asset/daml/blob/c33140cf88f80b3f212fd2a58b85d33e2e3f9711/compiler/damlc/tests/daml-test-files/NumericRound.daml
Thank you

**Comment by lehins** [comment]
Round half up cannot result in `-0.15` -> `-0.2`, because that is down!

`-0.15` is bigger `-0.2` or put in other words `-0.2` is below (not up) when compared to `-0.15`

What you are describing is round half away from zero. 

Also IEEE_754 is floating point number standard, which has nothing to do with decimal numbers, except that they look similar.

Also Python also does the right thing:

```python
>>> import decimal
>>> decimal.Decimal('-0.15').quantize(decimal.Decimal('0.1'), decimal.ROUND_HALF_UP)
Decimal('-0.2')
```

FYI Haskell uses round-half-even and safe-decimal provides many varieties of rounding. If you find one missing that you need, I'll be happy to accept a PR

Closing as incorrect. Feel free to reopen if you think I misunderstood you somehow.

**Comment by lehins** [comment]
Just realized that python does do what you describe, sorry it is really late for me. I'll look at it better when I wake up.

**Comment by symbiont-alexey-babkin** [comment]
No problems, thanks. 
So are you ok if I make a PR adding these two strategies?
https://en.wikipedia.org/wiki/Rounding#Round_half_towards_zero 
https://en.wikipedia.org/wiki/Rounding#Round_half_away_from_zero
and name them `RoundHalfTowardsZero` and `RoundHalfAwayFromZero`, correspondingly?

**Comment by lehins** [comment]
Now, with the fresh mind I do remember vividly discussing that problem exactly: (see my https://github.com/fpco/safe-decimal/issues/3#issuecomment-581060713).

> So are you ok if I make a PR adding these two strategies?

I am always in favor of such PRs. It should be pretty straightforward to follow by example, if you decide to make such contribution. Also if you can add some documentation about these discrepancies you discovered about comparison to other languages it would be a great bonus.

For consistency sake, however, names should be `RoundHalfToZero` (there is already `RoundToZero`)  and `RoundHalfFromZero`

**Comment by lehins** [comment]
Fixed in #7
